pkg/controller/loadbalancer/controller.go

```diff
@@ -175,9 +176,13 @@ func (h *Handler) ensureVMLoadBalancer(lbCopy, lb *lbv1.LoadBalancer) (*lbv1.Loa
 	if err != nil {
 		return lb, err
 	}
-	lbCopy.Status.BackendServers = getServerAddress(servers)
+	lbCopy.Status.BackendServers = getServerAddress(servers.GetBackendServers())
 	if len(lbCopy.Status.BackendServers) == 0 {
-		return lb, errNoRunningBackendServer
+		if servers.GetMatchedBackendServerCount() == 0 {
+			return lb, errNoRunningBackendServer
+		}
+		// there are matched backend servers, but none of them have IP
+		return lb, errAllBackendServersNoIP
```

Comment:
I am not familiar with the whole logic, seems to me that the criteria for no ip error is if the error is not no running backend server, I thought no running backend server is more generic than server having no IP?

Reply:
either, 

and the current design leaves the room to expand, as we might filter more, and the return error processing can be expanded to switch case them

```
	matchedRunningBackendServerCount int // the matched backend server count
	withAddressBackendServerCount    int // = len(Servers) for now, but can be other value if the Servers are further filtered
```